deletionwatcher.py

```diff
@@ -17,80 +20,110 @@
 
 # noinspection PyClassHasNoInit,PyBroadException,PyMethodParameters
 class DeletionWatcher:
-    @classmethod
-    def update_site_id_list(self):
-        soup = BeautifulSoup(requests.get("https://meta.stackexchange.com/topbar/site-switcher/site-list").text,
-                             "html.parser")
-        site_id_dict = {}
-        for site in soup.findAll("a", attrs={"data-id": True}):
-            site_name = urlparse(site["href"]).netloc
-            site_id = site["data-id"]
-            site_id_dict[site_name] = site_id
-        GlobalVars.site_id_dict = site_id_dict
+    def __init__(self):
+        DeletionWatcher.update_site_id_list()
+
+        self.socket = websocket.create_connection("wss://qa.sockets.stackexchange.com/")
+        self.posts = {}
+
+        if os.path.exists("deletionIDs.p"):
+            with open("deletionIDs.p", "rb") as fh:
+                for post in DeletionWatcher._check_batch(pickle.load(fh)):
+                    self.subscribe(post, pickle=False)
+
+                self._save()
+
+        threading.Thread(name="deletion watcher", target=self._start, daemon=True).start()
+
+    def _start(self):
+        while True:
+            msg = self.socket.recv()
+
+            if msg:
+                msg = json.loads(msg)
+                action = msg["action"]
+
+                if action == "hb":
+                    self.socket.send("hb")
+                else:
+                    data = json.loads(msg["data"])
+
+                    if data["a"] == "post-deleted":
+                        try:
+                            post_id, _, post_type, post_url, callbacks = self.posts[action]
+                            del self.posts[action["action"]]
+
+                            if not post_type == "answer" or ("aId" in data and str(data["aId"]) == post_id):
+                                self.socket.send("-" + action)
+                                Tasks.do(metasmoke.Metasmoke.send_deletion_stats_for_post, post_url, True)
+
+                                for callback, max_time in callbacks:
+                                    if callback and (not max_time or time.time() < max_time):
+                                        callback()
+                        except KeyError:
+                            pass
+
+    def subscribe(self, post_url, callback=None, pickle=True, timeout=None):
+        post_id, post_site, post_type = fetch_post_id_and_site_from_url(post_url)
+
+        if post_site not in GlobalVars.site_id_dict:
+            return
```

Comment:
Should we do something special here, like post to chat or otherwise panic? I'm assuming this isn't meant to happen often.